pkgs/development/libraries/java/swt/default.nix

```diff
@@ -49,37 +63,84 @@ in stdenv.mkDerivation rec {
     webkitgtk
   ];
 
-  NIX_LFLAGS = toString
-    ((map (x: "-L${lib.getLib x}/lib") [ xorg.libX11 pango gdk-pixbuf glib ]) ++
-      [ "-lX11" "-lpango-1.0" "-lgdk_pixbuf-2.0" "-lglib-2.0" ]);
+  patches = [ ./awt-libs.patch ./gtk-libs.patch ];
+
+  prePatch = ''
+    # clear whitespace from makefiles (since we match on EOL later)
+    sed -i 's/ \+$//' ./*.mak
+  '';
+
+  postPatch = let makefile-sed = builtins.toFile "swt-makefile.sed" (''
```

Comment:
Have you considered applying a patch and suggesting it upstream?

Reply:
Yes, but I figured upstream submissions could be figured out after our SWT was up-to-date first. I don't think 4.5 is still maintained. My original idea was to upgrade SWT after this was merged first, to keep each PR easier to follow.

Reply:
I tried to update SWT once..

https://github.com/NixOS/nixpkgs/pull/75609

Reply:
Do you still plan to update SWT now that this PR has been merged? @bb010g